### ISSUE TITLE
Convert drivers using PCI to latest MAME PCI interface.

### ISSUE THREAD

**Issue by andrew-gardner** [mamedev/mame#391]
Seattle, Vegas, Model3, and more drivers hook their PCI devices together using a legacy method.  Convert all drivers with PCI to the hookup style of drivers/iteagle.c.

**Comment by MoochMcGee** [comment]
This interface is incomplete for the purposes of at586, as there are devices required by at586 which need to be able to hook up custom registers in the configuration space. The new interface can't do this as far as I can tell.

**Comment by galibert** [comment]
Huh?  The configuration memory map is virtual for that reason.  Look at
i82875p for instance, it adds a bunch of registers.

  OG.

**Comment by angelosa** [comment]
Got an hard time understanding what's "Legacy" PCI, nowadays, my best guess is anything prefixed by L (lpci) therefore any class in [lpci.h](https://github.com/mamedev/mame/blob/master/src/devices/bus/lpci/pci.h) should be marked with the` [[deprecated("use non-legacy PCI device")]` marker?

**Comment by angelosa** [comment]
Break down OP message with tasks, marked drivers as resolved where `#include "machine/lpci.h"` is not found with current tree.

**Comment by angelosa** [comment]
Errata: there are THREE PCI bus implementations, the proper one is actually in [src/devices/machine/pci.cpp](https://github.com/mamedev/mame/blob/master/src/devices/machine/pci.cpp)

**Comment by angelosa** [comment]
Added model3.cpp: does its own PCI bus implementation, which for all purposes fits here.

**Comment by Hydreigon223** [comment]
Mind adding any drivers using the k033906.cpp PCI bridge device. The three drivers that use this are gticlub.cpp, nwk-tr.cpp and hornet.cpp.

**Comment by angelosa** [comment]
Good call, added. Also I'll change directory entry over the new format later on.

**Comment by angelosa** [comment]
Converted misc\comebaby.cpp to base PCI driver in PR #10830 , it wasn't doing anything useful to warrant using the legacy hookup. I'll deal with full conversion later since being i440bx.

**Comment by angelosa** [comment]
Updated OP message with easy to track base chipsets for x86 based systems.

**Comment by angelosa** [comment]
Updated OP with the (mostly) i440bx additions from [PR#11037](https://github.com/mamedev/mame/pull/11037).
As for the others:
- queen.cpp can be converted to stub PCI driver (as per voyager.cpp), unless anybody wants to tackle VIA architecture. That will be needed by calchase.cpp anyway;
- taitowlf.cpp can now be completely wiped off, since we have everything necessary to emulate a base PCTX-LA including the [Winbond Super I/O](https://github.com/mamedev/mame/commit/892cfe410bc0d3ba431ed3736816d428c26e3aac) and the onboard [ATI Rage II](https://github.com/mamedev/mame/commit/9de67ba5956f21cab6df8518835089bdf96548ee). pf2012 parallel/game port loading is special but I guess we cross that bridge after we have a solid i82439TX and not the other way around.

**Comment by angelosa** [comment]
- `konami/viper.cpp` should be easy now, given its limited PCI use beyond Voodoo itself.
- `funworld/photoply.cpp` tracked by `sis85c496` in https://github.com/mamedev/mame/blob/master/src/mame/pc/pcipc_sis.cpp
- `misc/savquest.cpp` tracked by `pciagp` in https://github.com/mamedev/mame/blob/master/src/mame/pc/pcipc.cpp

**Comment by angelosa** [comment]
Edited OP with remaining blanks (in particular PPC chipset targets).

**Comment by angelosa** [comment]
https://github.com/mamedev/mame/commit/6e1f17f676b772944049d5bfebda82bf51b228d4 takes care of `funworld/photoply.cpp`

**Comment by angelosa** [comment]
https://github.com/mamedev/mame/commit/a7940b351f6ec567aadf757a107d1aaf83e1a6d2 crosses `misc/savquest.cpp`.
